src/cache.js

```diff
@@ -0,0 +1,131 @@
+ * @private
+ * @param  {object} props
+ *         Properties describing the player record to cache.
+ *
+ * @param  {string} props.playerId
+ *         A player ID.
+ *
+ * @param  {string} props.embedId
+ *         An embed ID.
+ *
+ * @param  {string} [props.accountId="*"]
+ *         An optional account ID. This is optional because when we search for
+ *         pre-existing players to avoid downloads, we will not necessarily
+ *         know the account ID.
+ *
+ * @return {string}
+ *         A key to be used in the script cache.
+ */
+const key = ({accountId, playerId, embedId}) => `${accountId || '*'}_${playerId}_${embedId}`;
+
+/**
+ * Add an entry to the script cache.
+ *
+ * @private
+ * @param  {object} props
+ *         Properties describing the player record to cache.
+ *
+ * @param  {string} props.playerId
+ *         A player ID.
+ *
+ * @param  {string} props.embedId
+ *         An embed ID.
+ *
+ * @param  {string} [props.accountId="*"]
+ *         An optional account ID. This is optional because when we search for
+ *         pre-existing players to avoid downloads, we will not necessarily
+ *         know the account ID. If not given, we assume that no script was
+ *         downloaded for this player.
+ *
+ * @return {string}
+ *         A key to be used in the script cache.
+ */
+const store = (props) => {
+  cache.set(key(props), props.accountId ? getUrl(props) : '');
+};
+
+/**
+ * Checks if the script cache has an entry.
+ *
+ * @private
+ * @param  {object} props
+ *         Properties describing the player record to cache.
+ *
+ * @param  {string} props.playerId
+ *         A player ID.
+ *
+ * @param  {string} props.embedId
+ *         An embed ID.
+ *
+ * @param  {string} [props.accountId="*"]
+ *         An optional account ID. This is optional because when we search for
+ *         pre-existing players to avoid downloads, we will not necessarily
+ *         know the account ID.
+ *
+ * @return {boolean}
+ *         Will be `true` if there is a matching cache entry.
+ */
+const has = (props) => cache.has(key(props));
+
+/**
+ * Gets a cache entry.
+ *
+ * @private
+ * @param  {object} props
+ *         Properties describing the player record to cache.
+ *
+ * @param  {string} props.playerId
+ *         A player ID.
+ *
+ * @param  {string} props.embedId
+ *         An embed ID.
+ *
+ * @param  {string} [props.accountId="*"]
+ *         An optional account ID. This is optional because when we search for
+ *         pre-existing players to avoid downloads, we will not necessarily
+ *         know the account ID.
+ *
+ * @return {string}
+ *
+ */
+const get = (props) => cache.get(key(props));
+
+/**
+ * Clears the cache.
+ */
+const clear = () => {
+  cache.clear();
+};
+
+/**
+ * Iterates over the cache.
+ *
+ * @param  {Function} fn
+ *         A callback function that will be called with a value and a key
+ *         for each item in the cache.
+ */
+const forEach = (fn) => {
+  cache.forEach(fn);
+};
+
+export {
```

Comment:
Should this just be the default exports? Doesn't seem like you would want to only export one of these.

Reply:
Are you suggesting attaching them to an object and exporting that as the default?